### PR TITLE
Add cloud init

### PR DESCRIPTION
**Pull request by firecyberice**
- remove device init
- add cloud-init
- change tests
- update rootfs (without default user)
- change default *-init config files

### CODE REVIEW

**Comment by StefanScherer** [comment]
Unfortunately Travis does not work for pull requests.
But we have added Circle https://circleci.com/gh/firecyberice/image-builder-rpi64/7

File: builder/test-integration/spec/hypriotos-image/cloud-init_spec.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+require 'spec_helper'
+
+describe package('cloud-init') do
+  it { should be_installed }
+end
+
+describe file('/boot/user-data') do
```
Comment (DieterReuter) [inline builder/test-integration/spec/hypriotos-image/cloud-init_spec.rb]: Should we also add a test for file `/boot/meta-data` even if it's empty?

Comment (StefanScherer) [reply builder/test-integration/spec/hypriotos-image/cloud-init_spec.rb]: Good idea as it is mandatory to make cloud-init read the user-data file.

File: builder/test-integration/spec/hypriotos-image/device-init_spec.rb [hunk 2]
```diff
@@ -1,28 +1,9 @@
...
-#   it { should be_installed }
-# end
-
-# describe command('dpkg -l device-init') do
-#   its(:stdout) { should match /ii  device-init/ }
-#   its(:stdout) { should match /0.1.8/ }
-#   its(:exit_status) { should eq 0 }
-# end
-
 describe file('/boot/device-init.yaml') do
```
Comment (DieterReuter) [inline builder/test-integration/spec/hypriotos-image/device-init_spec.rb]: Better remove this test for file '/boot/device-init.yaml' when the file is removed from the image.

File: versions.config [hunk 3]
```diff
@@ -1,6 +1,6 @@
 # config vars for the root file system
-HYPRIOT_OS_VERSION="v1.1.1"
-ROOTFS_TAR_CHECKSUM="5698b4f7363316d25ce735024836b499a4306802518b81d7231d457bbefd4f53"
+HYPRIOT_OS_VERSION="v1.2.0"
```
Comment (DieterReuter) [inline versions.config]: @firecyberice could you please update to v1.2.1, just to get the very latest version in.

File: builder/test-integration/spec/hypriotos-image/device-init_spec.rb [hunk 4]
```diff
@@ -1,28 +1,9 @@
...
-  it { should be_file }
-  its(:content) { should match /hostname: / }
+  it { should_not be_file }
 end
 
 describe file('/usr/local/bin/device-init') do
```
Comment (DieterReuter) [inline builder/test-integration/spec/hypriotos-image/device-init_spec.rb]: Remove this test as well.

File: builder/test-integration/spec/hypriotos-image/docker_spec.rb [hunk 5]
```diff
@@ -72,13 +72,6 @@
   it { should be_owned_by 'root' }
 end
 
-describe file('/etc/bash_completion.d/docker') do
```
Comment (DieterReuter) [inline builder/test-integration/spec/hypriotos-image/docker_spec.rb]: This should test should be remove only until the correct docker-ce package for Debian Stretch can be installed. So maybe we should at least comment this test out and mark it as a todo item.

Comment (StefanScherer) [reply builder/test-integration/spec/hypriotos-image/docker_spec.rb]: It is cleaner to remove it. When it's part of an upstream package we do not have to test the upstream package in every detail.

Bash completion works for docker cli command with the current DEB package, but I don't know where the bash completion file is.

Comment (DieterReuter) [reply builder/test-integration/spec/hypriotos-image/docker_spec.rb]: Ok, agreed.

Comment (firecyberice) [reply builder/test-integration/spec/hypriotos-image/docker_spec.rb]: Should I remove the tests for docker completely except the first ones for checking the package installation and version? I would only keep theese lines: https://github.com/firecyberice/image-builder-rpi64/blob/add-cloud-init/builder/test-integration/spec/hypriotos-image/docker_spec.rb#L1-L13

Comment (DieterReuter) [reply builder/test-integration/spec/hypriotos-image/docker_spec.rb]: Not too much changes right now please, we can discuss this later when we redesign the complete build and testing processes.

File: builder/files/boot/user-data [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+#
+hostname: black-pearl
+manage_etc_hosts: true
+
+users:
+  - name: pirate
+    gecos: "Hypriot Pirate"
+    sudo: ALL=(ALL) NOPASSWD:ALL
+    shell: /bin/bash
+    groups: users,docker,video
```
Comment (DieterReuter) [inline builder/files/boot/user-data]: We should remove group `video` as we build a server OS only.

**Comment by DieterReuter** [comment]
LGTM, image booted and works great!
